src/account_manager/operations/stronghold.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright 2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use iota_client::secret::SecretManager;
+
+use crate::account_manager::AccountManager;
+
+impl AccountManager {
+    /// Sets the Stronghold password
+    pub async fn set_stronghold_password(&self, password: &str) -> crate::Result<()> {
+        if let SecretManager::Stronghold(stronghold) = &mut *self.secret_manager.write().await {
+            stronghold.set_password(password).await;
+            let result = stronghold.read_stronghold_snapshot().await;
+            if let Err(err) = result {
+                if let iota_client::Error::StrongholdProcedureError(ref err_msg) = err {
+                    if !err_msg.contains("IOError") {
```

Comment:
Hmm I don't like this, why is it not possible to do this at type level ?

Reply:
The error there is just a String, maybe can be changed in the stronghold part

Reply:
The String is already returned on Stronghold level https://github.com/iotaledger/stronghold.rs/blob/b7fb5e9453883b3ff23d935e4e581517c42c098c/client/src/actors/client.rs#L113 and not in iota.rs
I think with the new Stronghold version it's not like this anymore and we will update to it at some point

Reply:
Then please at least leave a TODO so we don't forget about it

Reply:
Added